### PR TITLE
algorithm class: rectangle

### PR DESCRIPTION
**Pull request by nyilynnhtwe**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [ ] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [ ] This pull request is all my own work -- I have not plagiarized.
- [ ] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [ ] All new JavaScript files are placed inside an existing directory.
- [ ] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [ ] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Geometry/Test/Rectangle.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import Rectangle from '../Rectangle'
+
+const rectangle = new Rectangle(4, 2)
+
+test('The area of a rectangle with length equal to 4 and width equal to 2', () => {
```
Comment (appgurueu) [inline Geometry/Test/Rectangle.test.js:5]: These tests are trivial; they don't need names.

File: Geometry/Rectangle.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  perimeter = () => {
+    return 2 * (this.length + this.width)
+  }
+
+  area = () => {
+    return this.length * this.width
+  }
+
+  diagonalLength = () => {
+    return Math.sqrt(Math.pow(this.length, 2) + Math.pow(this.width, 2))
```
Comment (appgurueu) [inline Geometry/Rectangle.js:25]: Why use `Math.pow` when JS has `**` for exponentiation?

File: Geometry/Rectangle.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * and check whether it is square or not.
+ * https://en.wikipedia.org/wiki/Rectangle
+ * @constructor
+ * @param {number} length - The length of the rectangle.
+ * @param {number} width  - The width  of the rectangle.
+ */
+
+export default class Rectangle {
+  constructor (length, width) {
+    this.length = length
```
Comment (appgurueu) [inline Geometry/Rectangle.js:12]: I think this is usually called "height"

**Review by appgurueu — COMMENTED** [review]
Pretty trivial. @raklaptudirm thoughts on accepting trivial algorithms / data structures?

**Comment by raklaptudirm** [comment]
I would say only to allow trivial algorithm implementations if they are able to showcase good usage of a large range of language features. This implementation doesn't seem to do that, and imo, has little educational value. I am in favor of closing this pr.

**Comment by nyilynnhtwe** [comment]
I saw the class "circle" and I wanted to implement for rectangle too. I can fixe the width to the height. Could you please consider to be merged?

**Comment by appgurueu** [comment]
I agree with rak that this is pretty trivial and has little educational value. You could consider:

1. Using more language features, such as getters
2. Making the class more useful by adding `x` and `y` fields, then implementing rectangle-rectangle intersection / collision and "is point in rectangle"

The latter is needed very often in game dev. It isn't terribly difficult to implement, but it is very useful and not as trivial as perimeter or area of a rectangle.

**Pull request closed without merging**